src/rules.jl

```diff
@@ -168,29 +170,34 @@ _abs_deriv(x) = signbit(x) ? -one(x) : one(x)
     :NaN, :(  (SpecialFunctions.bessely($ν - 1, $x) - SpecialFunctions.bessely($ν + 1, $x)) / 2  )
 @define_diffrule SpecialFunctions.besselk(ν, x)   =
     :NaN, :( -(SpecialFunctions.besselk($ν - 1, $x) + SpecialFunctions.besselk($ν + 1, $x)) / 2  )
+@define_diffrule SpecialFunctions.besselkx(ν, x)  =
+    :NaN, :( -(SpecialFunctions.besselkx($ν - 1, $x) + SpecialFunctions.besselkx($ν + 1, $x)) / 2 + SpecialFunctions.besselkx($ν, $x)  )
 @define_diffrule SpecialFunctions.hankelh1(ν, x)  =
     :NaN, :(  (SpecialFunctions.hankelh1($ν - 1, $x) - SpecialFunctions.hankelh1($ν + 1, $x)) / 2  )
+@define_diffrule SpecialFunctions.hankelh1x(ν, x) =
+    :NaN, :(  (SpecialFunctions.hankelh1x($ν - 1, $x) - SpecialFunctions.hankelh1x($ν + 1, $x)) / 2 - im * SpecialFunctions.hankelh1x($ν, $x)  )
```

Comment:
I always assumed that DiffRules only deals with functions with real arguments and real outputs - but `hankelh1x`, `hankelh2x`, and also `hankelh1` and `hankelh2` for which a rule is already defined output complex numbers. So complex numbers are supported officially? It seems based on the derivatives for `conj` and `adjoint` that you added in https://github.com/JuliaDiff/DiffRules.jl/pull/54 you also thought that DiffRules only deals with real numbers @mcabbott?